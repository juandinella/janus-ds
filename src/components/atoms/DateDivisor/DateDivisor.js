import isPropValid from '@emotion/is-prop-valid'
import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'
import styles from './DateDivisor.styles'
import Text from '../Text'

/**
 * DateDivisor
 */

const shouldForwardProp = (prop) => isPropValid(prop)
const StyledDateDivisor = styled.div.withConfig({shouldForwardProp})`${styles}`

const DateDivisor = ({ id, children }) => {
  return (
    <StyledDateDivisor id={id} data-testid={id}>
      <Text data-testid={`${id}-children`} type='span' size='xs'>{children}</Text>
    </StyledDateDivisor>
  )
}

DateDivisor.displayName = 'DateDivisor'

DateDivisor.propTypes = {
  /** Component id */
  id: PropTypes.string,
  children: PropTypes.node.isRequired,
}

DateDivisor.defaultProps = {}

export default DateDivisor
