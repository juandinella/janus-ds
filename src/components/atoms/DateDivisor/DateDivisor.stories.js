import DateDivisor from './DateDivisor'

import {
  getTemplate,
} from '../../../helpers/storybook'

const Template = getTemplate(DateDivisor)


export default {
  title: 'Atoms/DateDivisor',
  component: DateDivisor,
  args: {
    children: 'Aug 3, 2023, 11:45 AM',
    id: 'test'
  },
  argTypes: {},
}

export const Default = Template.bind({})

