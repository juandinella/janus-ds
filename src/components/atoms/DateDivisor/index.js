import DateDivisor from './DateDivisor'

export default DateDivisor
