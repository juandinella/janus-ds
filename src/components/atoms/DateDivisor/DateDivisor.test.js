import initStoryshots from '@storybook/addon-storyshots'
import { render, screen, cleanup } from '@testing-library/react'
import React from 'react'
import DateDivisor from './DateDivisor'

initStoryshots({
  storyKindRegex: /^Atoms\/DateDivisor$/,
})

const componentProps = {
  id: 'unique_id',
  children: 'Aug 3, 2023, 11:45 AM',
}
afterEach(cleanup)

describe('<DateDivisor />', () => {
  test('should render component correctly', () => {
    render(<DateDivisor {...componentProps} />)
    const divisor = screen.getByTestId(componentProps.id)
    expect(divisor).toBeInTheDocument()
  })

  test('should render children correctly', () => {
    render(<DateDivisor {...componentProps} />)
    const childrenElement = screen.getByTestId(`${componentProps.id}-children`)
    expect(childrenElement.textContent).toEqual(componentProps.children)
  })
})
