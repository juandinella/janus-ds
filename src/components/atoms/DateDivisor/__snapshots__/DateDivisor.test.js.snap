// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/DateDivisor Default 1`] = `
<div
  style={
    {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "position": "relative",
    }
  }
>
  <div
    className="sc-10r74t6-0 fNgNDf"
    data-testid="test"
    id="test"
  >
    <span
      className="sc-81qx22-0 btgklu"
      color="neutral-darker"
      size="xs"
    >
      Aug 3, 2023, 11:45 AM
    </span>
  </div>
</div>
`;
